From 39a415e8d8fadfdf1478fd45749edb6b76364994 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Tue, 15 Jun 2010 11:02:05 +1000
Patch-mainline: 2.6.35-rc4
References: fate#310916
Git-commit: 1c48bc5f71cd7783e19fb8d9462be53f829be177
Subject: [PATCH 2366/2588] agp: add no warn since we have a fallback to
 vmalloc paths

also drop the NORETRY we can probably nearly always satisfy order 1 allocs now,
and again the vmalloc path is there.

Signed-off-by: Dave Airlie <airlied@redhat.com>
(cherry picked from commit 1c48bc5f71cd7783e19fb8d9462be53f829be177)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/char/agp/generic.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/char/agp/generic.c b/drivers/char/agp/generic.c
index 0a4278c..ff7649e 100644
--- a/drivers/char/agp/generic.c
+++ b/drivers/char/agp/generic.c
@@ -98,7 +98,7 @@ void agp_alloc_page_array(size_t size, struct agp_memory *mem)
 	mem->pages = NULL;
 
 	if (size <= 2*PAGE_SIZE)
-		mem->pages = kmalloc(size, GFP_KERNEL | __GFP_NORETRY);
+		mem->pages = kmalloc(size, GFP_KERNEL | __GFP_NOWARN);
 	if (mem->pages == NULL) {
 		mem->pages = vmalloc(size);
 	}
-- 
1.7.6

